src/target/cortexa.c

```diff
@@ -317,6 +330,54 @@ static size_t create_tdesc_cortex_a(char *buffer, size_t max_len)
 	return (size_t)total;
 }
 
+typedef struct bitfield_entry {
+	char *desc;
+	uint8_t bitnum;
+} bitfields_lut_s;
+
+static const bitfields_lut_s cortexa_dbg_dscr_lut[] = {
+	{"HALTED", 0U},
+	{"RESTARTED", 1U},
+	{"SDABORT_l", 6U},
+	{"ADABORT_l", 7U},
+	{"UND_l", 8U},
+	{"FS", 9U},
+	{"ITRen", 13U},
+	{"HDBGen", 14U},
+	{"MDBGen", 15U},
+	{"InstrCompl_l", 24U},
+	{"PipeAdv", 25U},
+	{"TXfull_l", 26U},
+	{"RXfull_l", 27U},
+	{"TXfull", 29U},
+	{"RXfull", 30U},
+};
+
+static void helper_print_bitfields(const uint32_t val, const bitfields_lut_s lut[], const size_t array_length)
+{
+	for (size_t i = 0; i < array_length; i++) {
+		if (val & (1U << lut[i].bitnum)) {
+			DEBUG_TARGET("%s ", lut[i].desc);
+		}
+	}
+};
```

Comment:
Extra `;` here - please remove

Reply:
Yes, that's totally extra. Maybe I put that there by inertia after fixing struct and array declarations/definitions.
I also took the liberty of preemptively dropping the `for` & `if` curly-braces enclosing the single statement.

Reply:
You are correct on the `if`, but consider the `for` to be a multi-line block and so need braces - it's technically correct as-is, but the style used elsewhere is for such a block to be written as
```c
for (;;) {
    if (cond)
        action;
}
```

Reply:
Partially backed out and force-pushed to comply with this.